source/codegen/generate_service.py

```diff
@@ -25,11 +25,44 @@ def generate_service_file(metadata, template_file_name, generated_file_suffix, g
   f.write(template.render(data=metadata))
   f.close()
 
+
+class AttributeValueExpander:
+  '''Wraps an _attribute_type_map of:
+  
+  group -> type -> attributes
+
+  and implements the metadata_mutations to add_attribute_values_enums and expand_attribute_function_value_param.
+  '''
+  def __init__(self, metadata):
+    self._metadata = metadata
+    self._attribute_type_map = {}
+
+    for group_name, attributes in common_helpers.get_attribute_groups(metadata).items():
+      attribute_enums_by_type = common_helpers.get_attribute_enums_by_type(attributes)
+      metadata_mutation.add_attribute_values_enums(metadata["enums"], attribute_enums_by_type, group_name)
+      self._attribute_type_map[group_name] = attribute_enums_by_type
+
+  def get_used_attribute(self, parameters):
```

Comment:
I believe the better name for this function would be `get_used_attribute_group_name`, because essentially that's what it is returning. Corresponding variable name where this function is called should be updated too.